[release] v7.0.0-alpha.0 (#10966)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.0.0-alpha.0

_Nov 9, 2023_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

- ✨ v7 Alpha release
- 🌍 Improve Czech (cs-CZ) locale

From https://github.com/mui/mui-x/pull/10875

#### Breaking changes

Types for `slots` and `slotProps` got renamed by removing the "Component" which is meaningless for charts.

> Except if you imported those types, to create a wrapper, you should not be impacted by this breaking change.

Reviewer: ```suggestion
Unless you imported those types, to create a wrapper, you should not be impacted by this breaking change.
```

> Here is an example of the renaming for the `<ChartsLegend />` component.

Reviewer: ```suggestion
Here is an example of the renaming for the `<ChartsTooltip />` component.
```

```diff
-ChartsTooltipSlotsComponent
+ChartsTooltipSlots

-ChartsTooltipSlotComponentProps
+ChartsTooltipSlotProps
```

### Data Grid

> 

Reviewer 2: The breaking changes for other packages might also be added the same way. (Please remove extra `\` from the end of each diff before applying the change)

```suggestion

#### Breaking changes

- The deprecated props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead. See [components section](/x/react-data-grid/components/) for more details.

- The print export will now only print the selected rows if there are any.
  If there are no selected rows, it will print all rows. This makes the print export consistent with the other exports.
  You can [customize the rows to export by using the `getRowsToExport` function](/x/react-data-grid/export/#customizing-the-rows-to-export).

- The `getApplyFilterFnV7` in `GridFilterOperator` was renamed to `getApplyFilterFn`.
  If you use `getApplyFilterFnV7` directly - rename it to `getApplyFilterFn`.

- The signature of the function returned by `getApplyFilterFn` has changed for performance reasons:

```diff
 const getApplyFilterFn: GetApplyFilterFn<any, unknown> = (filterItem) => {
   if (!filterItem.value) {
     return null;
   }

   const filterRegex = new RegExp(escapeRegExp(filterItem.value), 'i');
-  return (cellParams) => {
-  const { value } = cellParams;
+  return (value, row, colDef, apiRef) => {
     return value != null ? filterRegex.test(String(value)) : false;
   };
 }
\```

- The `getApplyQuickFilterFnV7` in `GridColDef` was renamed to `getApplyQuickFilterFn`.
  If you use `getApplyQuickFilterFnV7` directly - rename it to `getApplyQuickFilterFn`.

- The signature of the function returned by `getApplyQuickFilterFn` has changed for performance reasons:

```diff
 const getGridStringQuickFilterFn: GetApplyQuickFilterFn<any, unknown> = (value) => {
   if (!value) {
     return null;
   }
   const filterRegex = new RegExp(escapeRegExp(value), 'i');
-  return (cellParams) => {
-    const { formattedValue } = cellParams;
+  return (value, row, column, apiRef) => {
+    let formattedValue = apiRef.current.getRowFormattedValue(row, column);
     return formattedValue != null ? filterRegex.test(formattedValue.toString()) : false;
   };
 };
\```
```

#### `@mui/x-data-grid@7.0.0-alpha.0`

- [DataGrid] Fix for error thrown when removing skeleton rows, after sorting is applied (#10807) @benjaminbialy
- [DataGrid] Fix: `undefined` slot value (#10937) @romgrk
- [DataGrid] Print selected rows by default (#10846) @cherniavskii
- [DataGrid] Remove deprecated `components` and `componentsProps` (#10911) @MBilalShafi
- [DataGrid] Remove legacy filtering API (#10897) @cherniavskii
- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10882) @michelengelen
- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen

> - [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka

Reviewer: Nitpick: Weren't we porting back the l10n changes during the v5 -> v6 transition? 🤔

Author: Yep ... I would expect that as well ... maybe that got lost somewhere?

Reviewer 3: Yes we were

Author: Maybe the current strategy is flawed.
Wouldn't it be better if we target `master` for bugfixes, translations, etc and before each release we merge `master` into `next`? Seems more feasible and it's harder to "lose" PRs

Reviewer: Your suggestion does have a lot of merit. 🤔 
What do others think @mui/x?

Reviewer 2: I think it will cause the PR not appearing in the next changelog like happened with https://github.com/mui/mui-x/pull/10543
The strategy of merging everything in next and `cherry-picking` the `l10n`, `bug fixes` and other things we decide to port seem the most suitable one, as there will be a single source of truth (next)
Regarding this specific change, we can push it on to the next master/v6 release, I think it won't hurt to release it one week later.

Reviewer 2: Nice suggestion, I am not against it, but I am still inclining towards providing more visibility of what's delivered in current release directly in the specific release page.

Since our changelogs comprises of changes on multiple packages, it might get cumbersome for the users to know what specific changes are coming from the master release for their specific package, and they might miss the ones coming from the master.

Another reason this might cause some issues is, there were cases in past when we delivered some changes on master which we did not deliver on next, (for example an API which we deprecated in the new major but needs some fixing in current major) so saying `Every change from [v6.18.1](#6.18.1) and the below.` would not possibly be true in that case. 🤷‍♂️

Author: Nice catch!
Well ... we can still go with the cherry-pick-to-master approach. We just need to be more careful to include all changes we want on master as well.

Not so sure about #10916 and #10913 as well, since those are still viable for `v6` as well.

Reviewer 2: If I remember correctly from what we discussed with the team the other day and the strategy we followed last year, from now on, we should only target important bug fixes (minor ones could possibly be skipped), localization changes and those things which are necessary for v6. Other low priority things like these recipes, new features, improvements and of coarse the breaking changes should be limited to the new major, basically to create some incentive for migration.
So, I'd vote not to port these recipes to v6.

Reviewer: > I think it will cause the PR not appearing in the next changelog like happened with #10543

Correct me if I'm wrong, but I think that the problem why individual changes are not picked up is the fact that we default to do a squash commit and merge.
If we decide to go with the proposal, then this particular PR would need to be merged regularly.
Doing a regular merge will include all the relevant merge commits in the branch and thus the generated changelog should be complete. 🤔 
WDYT?

Author: That should do the trick! 
And since this is only done occasionally it is only a slight deviation from the normal process!

I am pretty confident that this could be the better approach, but we'll see what the others have to say about it

Reviewer 2: > Another reason this might cause some issues is, there were cases in past (and could be in the future) when we delivered some changes on master which we did not deliver on next, (for example a deprecated API which we removed in the new major but needs some fixing in current major) so saying Every change from [v6.18.1](#6.18.1) and the below. would not possibly be true in that case.

Even if we manage to pick up the changes individually (it will include the release PR too I guess 😄), wouldn't ⬆️ still be a problem? 🤔

Author: I think we cannot circumvent manual filtering and adjustments in any case 🤷🏼

Reviewer: So, to summarize.
It seems that we are leaning toward sticking with the current approach of merging everything to `next` and cherry-picking manually only what's necessary.
If the problem arises once more - let's re-evaluate.
Because, as we've discussed, both approaches have their cons. Having the flexibility of choosing what we actually need sounds nice as well. 👌 
Correct me, if I made the wrong assumption? ℹ️

Reviewer 2: I think yes, that's what we'll be adapting to as of now, right @michelengelen?

Author: I guess that's the most reasonable approach for now! 👍🏻

#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

> - [DataGridPro] Autosize Columns - Headers being cut off (#10666) @gitstart

Reviewer 2: ```suggestion
- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
```

> 

Reviewer 2: It was merged along with master merge into next [PR](https://github.com/mui/mui-x/pull/10929/files), so no separate PR for this on next (my bad)
```suggestion
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.0.0-alpha.0

_Nov 9, 2023_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

> - ✨ v7 Alpha release
> - 🌍 Improve Czech (cs-CZ) locale

Reviewer 2: ```suggestion
- 🚀 First v7 alpha release
- ✨ Fix aggregation label not showing when `renderHeader` is used (#10961) @cherniavskii
- 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/) published
- 💫 New recipes added for the data grid
- 📈 `<ChartsReferenceLine />` component is now available
- 🌍 Improve Czech (cs-CZ) locale
- 🐞 Bugfixes
- 📚 Documentation improvements
```

- [DataGrid] Remove deprecated `components` and `componentsProps` (#10911) @MBilalShafi
- [DataGrid] Remove legacy filtering API (#10897) @cherniavskii
- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10882) @michelengelen
- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka

#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:

- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii

### Date Pickers

> 

Reviewer 2: I think we can add the breaking changes against #10700 here, someone from @mui/explore team, kindly check if the wording is correct?
```suggestion

#### Breaking changes

- The props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead.

```

Author: @flaviendelangle / @LukasTy gentle ping! 🔔

Reviewer: I've updated this suggestion and provided a tiny suggestion on the grid part as well. 🙈

- [DataGrid] Remove legacy filtering API (#10897) @cherniavskii
- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10882) @michelengelen
- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka

#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:

- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii

### Date Pickers

#### `@mui/x-date-pickers@7.0.0-alpha.0`

> - [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte

Reviewer: ```suggestion
- [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte
- [pickers] Refine `referenceDate` behavior in views (#10863) @LukasTy
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.0.0-alpha.0

_Nov 9, 2023_

We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

- 🚀 First v7 alpha release
- ✨ Fix aggregation label not showing when `renderHeader` is used (#10961) @cherniavskii
- 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/) published
- 💫 New recipes added for the data grid
- 📈 `<ChartsReferenceLine />` component is now available
- 🌍 Improve Czech (cs-CZ) locale
- 🐞 Bugfixes
- 📚 Documentation improvements

### Data Grid

#### Breaking changes

> - The deprecated props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead. See [components section](/x/react-data-grid/components/) for more details.

Reviewer: ```suggestion
- The deprecated `components` and `componentsProps` props have been removed. Use `slots` and `slotProps` instead. See [components section](/x/react-data-grid/components/) for more details.
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.0.0-alpha.0

> _Nov 9, 2023_

Author: ```suggestion
_Nov 10, 2023_
```

> We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

Reviewer: ```suggestion
We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:
```

- 🚀 First v7 alpha release
- ✨ Fix aggregation label not showing when `renderHeader` is used (#10961) @cherniavskii
- 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/) published
- 💫 New recipes added for the data grid
- 📈 `<ChartsReferenceLine />` component is now available

> - 🌍 Improve Czech (cs-CZ) locale

Reviewer: ```suggestion
- 🌍 Add Basque (eu) locale, improve Czech (cs-CZ) and Spanish (es-ES) locales
```

- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka

#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:

- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii

### Date Pickers

#### Breaking changes

- The deprecated `components` and `componentsProps` props have been removed. Use `slots` and `slotProps` instead.

#### `@mui/x-date-pickers@7.0.0-alpha.0`

- [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte
- [pickers] Refine `referenceDate` behavior in views (#10863) @LukasTy

> - [pickers] Remove `components` and `componentsProps` props (#10700) @alexfauquette

Reviewer: ```suggestion
- [pickers] Remove `components` and `componentsProps` props (#10700) @alexfauquette
- [l10n] Add Basque (eu) locale and improve Spanish (es-ES) locale (#10819) @lajtomekadimon
```

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 7.0.0-alpha.0

_Nov 10, 2023_

> We'd like to offer a big thanks to the 12 contributors who made this release possible. Here are some highlights ✨:

Reviewer 4: I'd also add a link to our v7 migration guide index page, do we have one?

```suggestion
We're thrilled to announce the first alpha release of our next major version, v7. 
This release introduces a few breaking changes, paving the way for the upcoming features like Pivoting and DateTimeRange.

A special shoutout to thank the 12 contributors who made this release possible. Here are some highlights ✨:
```

Author: We do not have a migration guide overview page, just single ones for each package. Should we introduce one?

Reviewer: Nice idea worth discussing.
What would you think of putting there? 🤔

- 🚀 First v7 alpha release
- ✨ Fix aggregation label not showing when `renderHeader` is used (#10961) @cherniavskii

> - 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/) published

Reviewer 4: ```suggestion
- 📘 Server side data source [early documentation](https://mui.com/x/react-data-grid/server-side-data/)
```

- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka

#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:

- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii

### Date Pickers

#### Breaking changes

- The deprecated `components` and `componentsProps` props have been removed. Use `slots` and `slotProps` instead.

#### `@mui/x-date-pickers@7.0.0-alpha.0`

> - [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte

Reviewer 5: ```suggestion
- [pickers] Escape non tokens words (#10400) @alexfauquette
- [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte
```

Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:

- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi

#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:

- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii

### Date Pickers

#### Breaking changes

- The deprecated `components` and `componentsProps` props have been removed. Use `slots` and `slotProps` instead.

#### `@mui/x-date-pickers@7.0.0-alpha.0`

- [pickers] Escape non tokens words (#10400) @alexfauquette
- [fields] Fix `MultiInputTimeRangeField` section selection (#10922) @noraleonte
- [pickers] Refine `referenceDate` behavior in views (#10863) @LukasTy
- [pickers] Remove `components` and `componentsProps` props (#10700) @alexfauquette
- [l10n] Add Basque (eu) locale and improve Spanish (es-ES) locale (#10819) @lajtomekadimon

> - [pickers] Add short weekdays token (#10988) @alexfauquette

Author: @alexfauquette its already in! 🎉
